boot/bootutil/src/swap_offset.c

```diff
@@ -307,23 +307,26 @@ static int app_max_sectors(struct boot_loader_state *state)
     uint32_t sz = 0;
     uint32_t sector_sz;
     uint32_t trailer_sz;
-    uint32_t first_trailer_idx;
+    uint32_t available_sectors_pri;
+    uint32_t available_sectors_sec;
+    uint32_t trailer_sectors = 0;
 
     sector_sz = boot_img_sector_size(state, BOOT_PRIMARY_SLOT, 0);
     trailer_sz = boot_trailer_sz(BOOT_WRITE_SZ(state));
-    first_trailer_idx = boot_img_num_sectors(state, BOOT_PRIMARY_SLOT);
 
     while (1) {
         sz += sector_sz;
+        ++trailer_sectors;
 
         if  (sz >= trailer_sz) {
             break;
         }
-
-        first_trailer_idx--;
     }
 
-    return first_trailer_idx;
+    available_sectors_pri = boot_img_num_sectors(state, BOOT_PRIMARY_SLOT) - trailer_sectors;
```

Comment:
To make sure, because this is swap_offset, the swap page is already included into this or should be -1 here to the equation for the swap page?

Reply:
there is no swap page in primary, only secondary, but the upgrade trailer data resides only in the primary, this is why secondary -1's for swap sector only and primary -'s the trailer size only